uui/components/datePickers/DatePickerBody.tsx

```diff
@@ -1,78 +1,170 @@
-import * as React from 'react';
+import React, { useEffect, useState } from 'react';
 import dayjs, { Dayjs } from 'dayjs';
 import updateLocale from 'dayjs/plugin/updateLocale.js';
-import { cx } from '@epam/uui-core';
-import { MonthSelection, YearSelection, DatePickerBodyBase, DatePickerBodyBaseProps, valueFormat, i18n } from '@epam/uui-components';
+import {
+    IControlled,
+    cx,
+} from '@epam/uui-core';
+import { MonthSelection, YearSelection } from '@epam/uui-components';
 import { DatePickerHeader } from './DatePickerHeader';
 import { Calendar } from './Calendar';
 import css from './DatePickerBody.module.scss';
+import { CommonDatePickerBodyProps, ViewType } from './types';
+import {
+    getNewMonth, uuiDatePickerBodyBase, valueFormat,
+} from './helpers';
 
-dayjs.extend(updateLocale);
-
-export interface DatePickerBodyProps extends DatePickerBodyBaseProps<string> {
-    /*
-    * A pure function that gets array of classes for styling a day for each day.
-    */
-    getDayCX?(day: Dayjs): string[];
-    /**
-     * A pure function that gets whether day is holiday for each day.
-     */
+export interface DatePickerBodyProps extends CommonDatePickerBodyProps, IControlled<string | null> {
     isHoliday?: (day: Dayjs) => boolean;
 }
 
+export type BodySettings = {
```

Comment:
lets rename interface to DatePickerBodyState and do we need to export it?

Reply:
renamed, do not export it

Reply:
this interface removed